src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,96 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie = ({ onAdd }: Props) => {
+  const [reset, setReset] = useState(0);
+  const initialForm = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+  const [form, setForm] = useState(initialForm);
+
+  function handleChange(field: keyof typeof form, value: string) {
+    setForm({ ...form, [field]: value });
+  }
+
+  function isFormValid() {
+    return (
+      form.title.trim() &&
+      form.imgUrl.trim() &&
+      form.imdbUrl.trim() &&
+      form.imdbId.trim()
+    );
+  }
```

Comment:
redundant function. just assign in constant in use it.

```suggestion
  const  isFormValid = 
      form.title.trim() &&
      form.imgUrl.trim() &&
      form.imdbUrl.trim() &&
      form.imdbId.trim()
    
```